packages/langium/src/references/references.ts

```diff
@@ -36,18 +36,19 @@ export class DefaultReferences implements References {
             if (assignment && nodeElem) {
                 // eslint-disable-next-line @typescript-eslint/no-explicit-any
                 const reference = (nodeElem as any)[assignment.feature] as unknown;
+
                 if (isReference(reference)) {
-                    const ref = reference.ref;
-                    if (ref && ref.$cstNode) {
-                        const targetNode = this.nameProvider.getNameNode(ref);
-                        if (!targetNode) {
-                            return ref.$cstNode;
-                        }
-                        else {
-                            return targetNode;
+                    return this.processReference(reference);
+                }
+                else if (reference instanceof Array) {
```

Comment:
Use `Array.isArray(reference)` instead of `reference instanceof Array`

Reply:
Will do